app/src/main/java/com/wut/indoornavigation/data/mesh/Mesh.java

```diff
@@ -0,0 +1,398 @@
+            graph.addVertex(vertex);
+
+            List<Vertex> vertices = findVerticesSetByFloorObjectSign(enumMap[x][y], floorNumber);
+
+            if (vertices != null) {
+                vertices.add(vertex);
+            }
+
+            return vertex;
+        }
+
+        if (enumMap[x][y] == FloorObject.BEACON) {
+            Point coordinates = new Point((float) x / 2, (float) y / 2, floorNumber);
+            List<Point> floorBeacons = beaconsDict.get(floorNumber);
+            if (floorBeacons == null) {
+                floorBeacons = new ArrayList<>();
+                beaconsDict.put(floorNumber, floorBeacons);
+            }
+            floorBeacons.add(coordinates);
+        }
+
+        return null;
+    }
+
+    @Nullable
+    private List<Vertex> findVerticesSetByFloorObjectSign(FloorObject floorObject, int floorNumber) {
+        List<Vertex> vertices = null;
+        if (floorObject == FloorObject.ROOM) {
+            if (destinationVerticesDict.containsKey(floorNumber)) {
+                vertices = destinationVerticesDict.get(floorNumber);
+            } else {
+                vertices = new ArrayList<>();
+                destinationVerticesDict.put(floorNumber, vertices);
+            }
+        }
+        if (floorObject == FloorObject.ELEVATOR) {
+            if (elevatorsVerticesDict.containsKey(floorNumber)) {
+                vertices = elevatorsVerticesDict.get(floorNumber);
+            } else {
+                vertices = new ArrayList<>();
+                elevatorsVerticesDict.put(floorNumber, vertices);
+            }
+        }
+        if (floorObject == FloorObject.STAIRS) {
+            if (stairsVerticesDict.containsKey(floorNumber)) {
+                vertices = stairsVerticesDict.get(floorNumber);
+            } else {
+                vertices = new ArrayList<>();
+                stairsVerticesDict.put(floorNumber, vertices);
+            }
+        }
+        return vertices;
+    }
+
+    @SuppressWarnings("ConstantConditions")
+    private void processNeighboursOfVertexOnFloor(Vertex vertex, int x, int y, FloorObject[][] enumMap, int floorNumber, boolean[][] visited, boolean[][] processedNeighbours, Graph graph) {
+        final int width = enumMap.length;
+        final int height = enumMap[0].length;
+
+        int startPosX = (x - 1 < 0) ? x : x - 1;
+        int startPosY = (y - 1 < 0) ? y : y - 1;
+        int endPosX = (x + 1 > width - 1) ? x : x + 1;
+        int endPosY = (y + 1 > height - 1) ? y : y + 1;
+
+        for (int rowNum = startPosX; rowNum <= endPosX; rowNum++) {
+            for (int colNum = startPosY; colNum <= endPosY; colNum++) {
+                Vertex v = processCell(rowNum, colNum, enumMap, floorNumber, visited, graph);
+
+                if (v != null) {
+                    double weight;
+                    if ((rowNum < x && colNum < y) || (rowNum > x && colNum < y) || (rowNum < x && colNum > y) || (rowNum > x && colNum > y)) {
```

Comment:
wydziel to do oddzielnej metody ktora zwraca boola i jej nazwa mowi co sprawdzasz